xbmc/interfaces/legacy/Addon.cpp

```diff
@@ -155,129 +158,131 @@ namespace XBMCAddon
       return ret;
     }
 
-    Settings* Addon::getSettings(unsigned int id /* = 0*/)
+    Settings* Addon::getSettings(unsigned int instance /* = 0*/)
     {
-      return new Settings(pAddon->GetSettings(id));
+      return new Settings(pAddon->GetSettings(instance));
     }
 
-    String Addon::getSetting(const char* id)
+    String Addon::getSetting(const char* id, unsigned int instance /* = 0*/)
     {
       return pAddon->GetSetting(id);
```

Comment:
NOTE: Have forgot give of "instance" here, but not updated now, as the support there maybe becomes removed again.